Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -919,35 +896,10 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
                 }
                 completionHandler?(nil)
             }
-        case .setSelectedServer(let selectedServer):
-            let serverSelectionMethod: NetworkProtectionServerSelectionMethod
-
-            switch selectedServer {
-            case .automatic:
-                serverSelectionMethod = .automatic
-            case .endpoint(let serverName):
-                serverSelectionMethod = .preferredServer(serverName: serverName)
-            }
-
-            Task { @MainActor in
-                if case .connected = connectionStatus {
-                    try? await updateTunnelConfiguration(environment: settings.selectedEnvironment, serverSelectionMethod: serverSelectionMethod, reassert: true)
-                }
-                completionHandler?(nil)
-            }
-        case .setSelectedLocation(let selectedLocation):
-            let serverSelectionMethod: NetworkProtectionServerSelectionMethod
-
-            switch selectedLocation {
-            case .nearest:
-                serverSelectionMethod = .automatic
-            case .location(let location):
-                serverSelectionMethod = .preferredLocation(location)
-            }
```

Comment:
Can you explain why this is all going away?  Admittedly I'm not looking to deeply into this and opting to ask as I want to avoid making (mistaken) assumptions.

Reply:
Yes sure. It’s essentially extracting this logic into the… class that has yet to be named ;). Let’s go with `ServerSelector`. I’ve been trying to avoid adding too much more logic to the PacketTunnelProvider, but this selection logic seemed closely related to the responsibilities of the `ServerSelector`. The extraction should allow it to all be unit tested too which I’ll do before moving this PR out of draft.